src/ucp/core/ucp_ep.c

```diff
@@ -1637,19 +1691,22 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
                 rma_config->max_put_bcopy = ucs_min(iface_attr->cap.put.max_bcopy,
                                                     rma_config->put_zcopy_thresh);
             }
-            if (iface_attr->cap.flags & UCT_IFACE_FLAG_PUT_SHORT) {
-                rma_config->max_put_short = ucs_min(iface_attr->cap.put.max_short,
-                                                    rma_config->max_put_bcopy);
-            }
 
             /* GET */
+            if (iface_attr->cap.flags & UCT_IFACE_FLAG_GET_SHORT) {
+                rma_config->max_get_short = iface_attr->cap.get.max_short;
+            } else {
+                rma_config->max_get_short = -1;
```

Comment:
previous approach required less code

Reply:
where is it avoided?

Reply:
it is avoided here:
https://github.com/openucx/ucx/blob/d8912b46f04243512a31250759a2271d6f626351/src/ucp/rma/rma_send.c#L255

before my fix, if `max_put_short` is 0, doing `ucp_ep_put_nb()` still uses PUT Short for a 0-byte message